components/inputs/TextInput.tsx

```diff
@@ -1,28 +1,48 @@
 import { clsx } from 'clsx'
-import React from 'react'
-import { TextInput as TextInputNative, TextInputProps } from 'react-native'
+import { forwardRef, ReactNode, useCallback, useRef } from 'react'
+import {
+  TextInput as TextInputNative,
+  TextInputProps as TextInputNativeProps,
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native'
 
-type Props = Omit<TextInputProps, 'editable'> & {
+import { useMultipleRefsSetter } from '@/hooks/useMultipleRefsSetter'
+
+export type TextInputProps = Omit<TextInputNativeProps, 'editable'> & {
   hasError?: boolean
   isDisabled?: boolean
+  leftIcon?: ReactNode
 }
 
 // TODO associate control with label
 // eslint-disable-next-line no-secrets/no-secrets
 // TODO multiline height on ios, inspiration?: https://stackoverflow.com/questions/35936908/numberoflines-textinput-property-not-working
 
-const TextInput = ({ hasError, isDisabled, ...rest }: Props) => {
-  return (
-    <TextInputNative
-      className={clsx('rounded border bg-white px-4 py-3', {
-        'border-divider focus:border-dark': !isDisabled && !hasError,
-        'border-negative': hasError && !isDisabled,
-        'border-divider bg-[#D6D6D6]': isDisabled,
-      })}
-      editable={!isDisabled}
-      {...rest}
-    />
-  )
-}
+const TextInput = forwardRef<TextInputNative, TextInputProps>(
+  ({ hasError, isDisabled, leftIcon, ...rest }, ref) => {
+    const localRef = useRef<TextInputNative>(null)
+    const refSetter = useMultipleRefsSetter(localRef, ref)
+
+    const handlePress = useCallback(() => {
+      localRef.current?.focus()
+    }, [])
+
+    return (
+      <TouchableWithoutFeedback onPress={handlePress}>
+        <View
+          className={clsx('flex-row rounded border bg-white px-4 py-3 g-3', {
+            'border-divider focus:border-dark': !isDisabled && !hasError,
+            'border-negative': hasError && !isDisabled,
+            'border-divider bg-[#D6D6D6]': isDisabled,
+          })}
+        >
+          <View className="h-6 w-6">{leftIcon ?? null}</View>
```

Comment:
Using ternary operator will prevent to render empty View:
```
{leftIcon ? <View className="h-6 w-6">{leftIcon}</View> : null}
```

That's how we usually handle conditional rendering.

Reply:
Please also provide `aria-hidden` prop to icon view, so it's hidden for screen readers. (at least that's what would I do in web)

Reply:
The first comment will probably solve also this new empty space on all text inputs:
![image](https://github.com/bratislava/paas-mpa/assets/19418224/8669ab66-af04-45d2-a34e-e4012e5cd144)